test/block.js

```diff
@@ -172,56 +139,57 @@ tape('alice does not replicate messages from bob, but carol does', function (t)
     })
   )
 
-  cont.para([
-    cont(alice.publish)(u.follow(carol.id)),
-    cont(bob.publish)({ type: 'post', text: 'hello' }),
-    cont(carol.publish)(u.follow(bob.id))
-  ])(function (err, r) {
-    t.error(err)
-    const recv = { alice: 0, carol: 0 }
-    carol.post(function (msg) {
-      recv.carol++
-      // will receive one message from bob and carol
-    }, false)
-
-    alice.post(function (msg) {
-      recv.alice++
-      // alice will only receive the message from carol, but not bob.
-      t.equal(msg.value.author, carol.id)
-    }, false)
-
-    carol.friends.get(function (err, g) {
-      t.error(err)
-      t.ok(g[carol.id][bob.id])
-    })
-
-    let n = 2
-    carol.connect(alice.getAddress(), cb)
-    carol.connect(bob.getAddress(), cb)
-
-    function cb (err, rpc) {
-      if (err) throw err
-      rpc.on('closed', next)
-    }
-    function next () {
-      if (--n) return
-      pull(
-        carol.createLogStream(),
-        pull.collect(function (err, ary) {
-          if (err) throw err
-          carol.getVectorClock(function (err, vclock) {
-            t.error(err)
-            t.equals(vclock[alice.id], 3)
-            t.equals(vclock[bob.id], 2)
-            t.equals(vclock[carol.id], 2)
-
-            t.equal(friends, 3, "carol's createFriendStream has 3 peers")
-            t.end()
-          })
-        })
-      )
-    }
+  await Promise.all([
+    pify(alice.publish)(u.follow(carol.id)),
+    pify(bob.publish)({ type: 'post', text: 'hello' }),
+    pify(carol.publish)(u.follow(bob.id))
+  ])
+
+  const recv = { alice: 0, carol: 0 }
+
+  // carol will receive one message from bob and carol
+  carol.post((msg) => recv.carol++, false)
+
+  // alice will only receive the message from carol, but not bob.
+  alice.post((msg) => {
+    recv.alice++
+    t.equal(msg.value.author, carol.id)
+  }, false)
+
+  const g = await pify(carol.friends.get)()
+  t.ok(g[carol.id][bob.id])
+
+  const [rpcCarolToAlice, rpcCarolToBob] = await Promise.all([
+    pify(carol.connect)(alice.getAddress()),
+    pify(carol.connect)(bob.getAddress()),
+  ])
+
+  await sleep(REPLICATION_TIMEOUT)
+
+  await pify(rpcCarolToAlice.close)(true)
+  await pify(rpcCarolToBob.close)(true)
+
+  // Drain Carol's full log
+  await new Promise((resolve, reject) => {
+    pull(
+      carol.createLogStream(),
+      pull.collect(function (err, ary) {
+        if (err) reject(err)
+        else resolve(ary)
+      }),
+    )
   })
+
+  const vclock = await pify(carol.getVectorClock)()
+  t.equals(vclock[alice.id], 3)
+  t.equals(vclock[bob.id], 2)
+  t.equals(vclock[carol.id], 2)
+
+  t.equals(recv.alice, 2)
+  t.equals(recv.carol, 3)
```

Comment:
Based on the comment where these are incremented I would have thought they would be alice: 1 and carol: 2? Is this because it continues the tests?

Reply:
Yeah, you're right, this is not supposed to be 2 and 3, but I just preserved the same behavior as before (I actually ran the previous tests and tried to check what these values were). You can confirm this by running tests in `master`. I think it's worth investigating, maybe it's something in ssb-db or maybe it's the `sbot.post`. Not sure what's the cause.

Reply:
I'll have a look

Reply:
Yeah so it is the tests that are overlapping. So in the last one, once carol connects to bob, she will get both the post but also bobs follow of alice from earlier. So I think it should be enough to fix the comments above the `carol.post` and `alice.post`